src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -4090,6 +4090,88 @@ public void Checked()
+            parsedArgs.Errors.Verify();
+            Assert.False(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable", "/langversion:8", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.True(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable-", @"/nullable", "/langversion:8", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.True(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable+", @"/nullable-", "/langversion:8", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.False(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable:", "/langversion:8", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify(Diagnostic(ErrorCode.ERR_BadSwitch).WithArguments("/nullable:"));
+            Assert.Null(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable+", "/langversion:7.3", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify(
+                // error CS8630: Invalid 'nullable' value: 'True' for C# 7.3. Please use language version 8.0 or greater.
+                Diagnostic(ErrorCode.ERR_NullableOptionNotAvailable).WithArguments("nullable", "True", "7.3", "8.0").WithLocation(1, 1)
+                );
+            Assert.True(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable-", "/langversion:7.3", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify(
+                // error CS8630: Invalid 'nullable' value: 'False' for C# 7.3. Please use language version 8.0 or greater.
+                Diagnostic(ErrorCode.ERR_NullableOptionNotAvailable).WithArguments("nullable", "False", "7.3", "8.0").WithLocation(1, 1)
+                );
+            Assert.False(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { @"/nullable", "/langversion:7.3", "a.cs" }, _baseDirectory);
+            parsedArgs.Errors.Verify(
+                // error CS8630: Invalid 'nullable' value: 'True' for C# 7.3. Please use language version 8.0 or greater.
+                Diagnostic(ErrorCode.ERR_NullableOptionNotAvailable).WithArguments("nullable", "True", "7.3", "8.0").WithLocation(1, 1)
+                );
+            Assert.True(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { "a.cs", "/langversion:8" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.Null(parsedArgs.CompilationOptions.Nullable);
+
+            parsedArgs = DefaultParse(new[] { "a.cs", "/langversion:7.3" }, _baseDirectory);
+            parsedArgs.Errors.Verify();
+            Assert.Null(parsedArgs.CompilationOptions.Nullable);
+        }
```

Comment:
>} [](start = 8, length = 1)

If I run `csc.exe /nullable+ /langversion:7.3 a.cs`, would I get two diagnostics (one from parsing command-line options and one from compilation diagnostics), or just one?

Reply:
> If I run csc.exe /nullable+ /langversion:7.3 a.cs, would I get two diagnostics (one from parsing command-line options and one from compilation diagnostics), or just one?

Just the one reported by the command line parser. Note that both places report errors originating from options.

---
In reply to: [224929329](https://github.com/dotnet/roslyn/pull/30479#discussion_r224929329) [](ancestors = 224929329)